### PR TITLE
Admin - Category - Update Category

### PR DESCRIPTION
**Pull request by at-vudang95**

### CODE REVIEW

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -6,5 +6,33 @@
 
 class CategoryController extends Controller
```
Comment (at-duyho) [inline app/Http/Controllers/CategoryController.php]: you should create constructor to use DI laravel.

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -6,5 +6,33 @@
...
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param int $id It is id of category need update
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $category = Category::find($id);
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: you should use findOrFail function

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -6,5 +6,33 @@
...
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request Request from client
+     * @param int                      $id      It is id of category need update
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $category = Category::find($id);
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: you should use findOrFail function.

File: app/Http/Controllers/CategoryController.php [hunk 4]
```diff
@@ -6,5 +6,33 @@
...
+        $category->name = $request->categoryName;
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: you should validate data.

**Review by ghost — CHANGES_REQUESTED** [review]
check my comment. thank you!

File: app/Http/Controllers/CategoryController.php [hunk 5]
```diff
@@ -2,9 +2,64 @@
...
+    private $category;
+
+    /**
+     * CategoryController constructor.
+     *
+     * @param Category $cate It is param input constructors
+     */
+    public function __construct(Category $cate)
+    {
+        $this->category = $cate;
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: -you should define clear variable: $cate => $category

File: app/Http/Controllers/CategoryController.php [hunk 6]
```diff
@@ -2,9 +2,64 @@
...
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request Request from client
+     * @param int                      $id      It is id of category need update
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $this->validate($request, [
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: -you should create a new Request class to validate data.
https://mattstauffer.co/blog/laravel-5.0-form-requests

File: app/Http/Controllers/CategoryController.php [hunk 7]
```diff
@@ -2,9 +2,64 @@
...
+            'name' => 'required|unique:categories|max:255',
+            'description' => 'required',
+        ], [
+            'name.required' => ' The category name field is required.',
+            'name.max' => ' The category name may not be greater than 255 characters.',
+            'name.unique' => ' The category name is existed.',
+            'description.required' => ' The description field is required.',
+        ]);
+        $cate =  $this->category->findOrFail($id);
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: -you should define clear variable: $cate => $category

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment and fix permission file (10664->100755)

File: app/Http/Controllers/CategoryController.php [hunk 8]
```diff
@@ -2,9 +2,56 @@
...
+     * @param int                      $id      It is id of category need update
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(CategoryRequest $request, $id)
+    {
+        $category =  $this->category->findOrFail($id);
+        $category->name = $request->name;
+        $category->description = $request->description;
+        $category->save();
```
Comment (ghost) [inline app/Http/Controllers/CategoryController.php]: -you should use: $this-> category->update($request->all());

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment. thank!

File: app/Http/Requests/CategoryRequest.php [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+         return [
+           'name' => 'required|unique:categories|max:255',
+           'description' => 'required',
```
Comment (ghost) [inline app/Http/Requests/CategoryRequest.php]: do description require?

Comment (at-vudang95) [reply app/Http/Requests/CategoryRequest.php]: No, I will remove.

**Review by ghost — APPROVED** [review]
LGTM

**Review by ghost — CHANGES_REQUESTED** [review]
I see the fatal error in this PR, please fix and please test on your local before assign reviewer

**Comment by at-vudang95** [comment]
I merged, fixed and tested in local. Please you check again it. Thank you!

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**